### PR TITLE
Misra C Compliances for files

### PR DESCRIPTION
**Pull request by NidhiSura**
Misra C compliances implemented for following files:
1. EEPROM.c
2. circularbuffer.c
3. crc.c
4. led.c
5. uart_default_cotrol.c

### CODE REVIEW

File: app/app_inc/crc.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ Copyright 2020 Samuel Ramrajkar
 extern "C" {
 #endif
 
-    uint16_t crc16_app(void* dptr, uint16_t len, uint16_t seed);
+    uint16_t crc16_app(uint8_t* dptr, uint16_t len, uint16_t seed);
```
Comment (sramrajkar) [inline app/app_inc/crc.h:30]: What was the reported MISRA error here?

File: app/app_src/circular_buffer.c [hunk 2]
```diff
@@ -60,7 +60,8 @@ void CircularBufferInit(CircularBufferContext *ctx, void *buf, size_t buf_size,
     ctx->write_pos = 0u;
     ctx->read_pos = 0u;
     ctx->element_size = element_size;
-    ctx->max_size = size - 1;
+    int sizeint = (int)size;
```
Comment (sramrajkar) [inline app/app_src/circular_buffer.c:63]: What was the error here? Also we should use unsigned int when signed arithmetic is not needed

File: app/app_src/circular_buffer.c [hunk 3]
```diff
@@ -70,53 +71,63 @@ void CircularBufferClear(CircularBufferContext *ctx) {
...
+    return retvar3;
 }
 
 int8_t CircularBufferPeek(const CircularBufferContext *ctx, size_t num,
                            void **elem) {
+    int retvar;
     size_t write_pos = ctx->write_pos;
     size_t read_pos = ctx->read_pos;
     size_t size = ((write_pos - read_pos) & ctx->max_size);
+    int sizeint2 = (int)size;
```
Comment (sramrajkar) [inline app/app_src/circular_buffer.c:113]: Use unsigned if signed arithmetic is not needed

File: app/app_src/circular_buffer.c [hunk 4]
```diff
@@ -70,53 +71,63 @@ void CircularBufferClear(CircularBufferContext *ctx) {
...
-    return 0;
-
-fail:
-    return -1;
+    retvar = 0;
+}
+    return retvar;
+   
+//fail:
+//    return -1;
```
Comment (sramrajkar) [inline app/app_src/circular_buffer.c:130]: delete if not needed

File: app/app_src/crc.c [hunk 5]
```diff
@@ -16,8 +16,9 @@ Copyright 2020 Samuel Ramrajkar
 #include "mcc.h"
 #include "crc.h"
 
-uint16_t crc16_app(void* dptr, uint16_t len, uint16_t seed){
-    uint8_t* ptr = (uint8_t*)dptr;
+uint16_t crc16_app(uint8_t* dptr, uint16_t len, uint16_t seed){
+    uint8_t* ptr = dptr;
+    //uint8_t* ptr = (uint8_t*)dptr;
```
Comment (sramrajkar) [inline app/app_src/crc.c:21]: delete this if not needed

File: app/app_src/crc.c [hunk 6]
```diff
@@ -26,14 +27,31 @@ uint16_t crc16_app(void* dptr, uint16_t len, uint16_t seed){
...
+        busybool = false;
+    }
+    else{
+        busybool = true;
+    }
+    
     for(uint16_t i = 0; i < len; i++ )
     {
-        while(CRCCON0bits.FULL); //Wait if the module is busy
+        while(fullbool){} //Wait if the module is busy
```
Comment (sramrajkar) [inline app/app_src/crc.c:49]: This is bug. Code will lock up here. same at line 54.
The original code is checking for hardware register flags we can replace this.

File: app/app_src/led.c [hunk 7]
```diff
@@ -51,19 +51,20 @@ void ledInit(void)
 }
 
 int8_t queueLedEvent(uint8_t LEDState,uint16_t duration)
-{
+{   int retvar1;
```
Comment (sramrajkar) [inline app/app_src/led.c:54]: create variable on new line

File: app/app_src/crc.c [hunk 8]
```diff
@@ -26,14 +27,31 @@ uint16_t crc16_app(void* dptr, uint16_t len, uint16_t seed){
     //CRC_Start();
     CRCCON0bits.EN = 1;
     CRCCON0bits.CRCGO = 1;
+    
+    bool fullbool;
```
Comment (sramrajkar) [inline app/app_src/crc.c:31]: create all variables at the start of the function. Just a style issue.

File: app/app_src/led.c [hunk 9]
```diff
@@ -99,7 +109,7 @@ void handle_led_events(void)
 void queue_serial_led_event(void)
 {
     queueLedEvent(0,400);
-    for(uint8_t i = 0; i < 2; i++)
+    for(int i = 0; i < 2; i++)
```
Comment (sramrajkar) [inline app/app_src/led.c:112]: use unsigned int. What was the error here?

File: app/app_src/uart_default_control.c [hunk 10]
```diff
@@ -23,6 +23,13 @@ enum UART_DEFAULT_STATE_T{
 enum UART_DEFAULT_STATE_T uart_default_state_var = UART_DEFAULT_INIT;
     
 void uart_default_engine(void){
+    bool getvaluebool;
+            if(BLEN_GetValue() == 0){
+                getvaluebool = false;
+            }
+            else{
+                getvaluebool =  true;
+            }
```
Comment (sramrajkar) [inline app/app_src/uart_default_control.c:32]: Not needed. was this for condition statement? Lets talk over a call and see what we can do to solve this issue

File: app/app_src/uart_default_control.c [hunk 11]
```diff
@@ -70,18 +81,24 @@ void uart_default_engine(void){
             break;
         case SWITCH_TO_CURRENT:
 #ifdef ATCOMM
-            set_parity(DATAEE_ReadByte_Platform(UARTParity));
-            set_uart_baud(DATAEE_ReadByte_Platform(UARTBaud));
+            {
+                int setpar1 = set_parity(DATAEE_ReadByte_Platform(UARTParity));
+                int setbaud2 = set_uart_baud(DATAEE_ReadByte_Platform(UARTBaud));
```
Comment (sramrajkar) [inline app/app_src/uart_default_control.c:86]: Do we need to have return values captured?

File: app/app_src/uart_default_control.c [hunk 12]
```diff
@@ -70,18 +81,24 @@ void uart_default_engine(void){
...
+            }
 #endif
 #ifdef MBRTU
-            set_parity(curent_parity);
-            set_uart_baud(current_baud_rate);
-            eMBInit( MB_RTU, mb_rtu_addr, 0, current_baud_rate, 
-                                             curent_parity);
+            {
+                uint8_t parity_status = (uint8_t)set_parity(curent_parity);
+                uint8_t uart_baud_status = (uint8_t)set_uart_baud(current_baud_rate);
```
Comment (sramrajkar) [inline app/app_src/uart_default_control.c:92]: Do we need to capture the returned values?

**Review by sramrajkar — CHANGES_REQUESTED** [review]
Some bugs are present

File: app/app_inc/crc.h (hunk 1 above)
Comment (NidhiSura) [reply app/app_inc/crc.h:30]: crc.c:17:20: [misra-c2012-11.5] A conversion should not be performed from pointer to void into pointer to object

File: app/app_src/circular_buffer.c (hunk 2 above)
Comment (NidhiSura) [reply app/app_src/circular_buffer.c:63]: circularbuffer.c:64:26: [misra-c2012-10.4] Both operands of an operator in which the usual arithmetic conversions are performed shall have the same essential type category

Yeah, I'll change it to an unsigned int.

File: app/app_src/uart_default_control.c (hunk 11 above)
Comment (atharva-18) [reply app/app_src/uart_default_control.c:86]: Yes, according to misra-c2012-17.7, the value returned by a non-void function must be used.

File: app/app_src/uart_default_control.c (hunk 12 above)
Comment (atharva-18) [reply app/app_src/uart_default_control.c:92]: Yes, according to misra-c2012-17.7, the value returned by a non-void function must be used.

**Comment by atharva-18** [comment]
@sramrajkar Please check the recent commits.

File: app/app_src/EEPROM.c [hunk 13]
```diff
@@ -66,82 +71,83 @@ void DATAEE_WriteByte_Platform(uint16_t addr, uint8_t data){
...
-    }
-    //Atleast one flag is set to sink data back to EEPROM
-    if(eeprom_write_flags.flag_bootload){
-        DATAEE_WriteByte_Platform(REQBootLoad, 0xAA);
-        eeprom_write_flags.flag_bootload = 0;
-    }
-    else if(eeprom_write_flags.flag_appkey){
-        for(uint8_t i = 0; i < sizeof(aes_key); i++){
-            DATAEE_WriteByte_Platform((AESkey0 + i),aes_key[i]);
+    if(eeprom_write_flags.flag_master > 0){
```
Comment (sramrajkar) [inline app/app_src/EEPROM.c:74]: Check for (0 != eeprom_write_flags.flag_master)

File: app/app_src/EEPROM.c [hunk 14]
```diff
@@ -66,82 +71,83 @@ void DATAEE_WriteByte_Platform(uint16_t addr, uint8_t data){
...
+        if(eeprom_write_flags.flag_bootload > 0){
+            DATAEE_WriteByte_Platform(REQBootLoad, 0xAA);
+            eeprom_write_flags.flag_bootload = 0;
         }
-        eeprom_write_flags.flag_appkey = 0;
-    }
-    else if(eeprom_write_flags.flag_netkey){
-        for(uint8_t i = 0; i < sizeof(net_key); i++){
-            DATAEE_WriteByte_Platform((NETkey0 + i),net_key[i]);
+        else if(eeprom_write_flags.flag_appkey > 0){
```
Comment (sramrajkar) [inline app/app_src/EEPROM.c:80]: Same as above

File: app/app_src/circular_buffer.c [hunk 15]
```diff
@@ -60,7 +60,8 @@ void CircularBufferInit(CircularBufferContext *ctx, void *buf, size_t buf_size,
     ctx->write_pos = 0u;
     ctx->read_pos = 0u;
     ctx->element_size = element_size;
-    ctx->max_size = size - 1;
+    uint8_t sizeint = (uint8_t)size;
+    ctx->max_size = sizeint - 1u;
```
Comment (sramrajkar) [inline app/app_src/circular_buffer.c:64]: What is the error here?

File: app/app_src/circular_buffer.c [hunk 16]
```diff
@@ -70,53 +71,59 @@ void CircularBufferClear(CircularBufferContext *ctx) {
...
+    else{
+        memcpy(val, &ctx->buf[ctx->read_pos * ctx->element_size],
            ctx->element_size);
 
-    ctx->read_pos = (ctx->read_pos + 1) & ctx->max_size;
+        ctx->read_pos = (ctx->read_pos + 1) & ctx->max_size;
 
-    return 0;
+        retvar3 = 0;
+        }
```
Comment (sramrajkar) [inline app/app_src/circular_buffer.c:103]: Indentation

File: app/app_src/circular_buffer.c [hunk 17]
```diff
@@ -70,53 +71,59 @@ void CircularBufferClear(CircularBufferContext *ctx) {
...
     size_t write_pos = ctx->write_pos;
     size_t read_pos = ctx->read_pos;
     size_t size = ((write_pos - read_pos) & ctx->max_size);
+    uint8_t sizeint2 = (uint8_t)size;
 
     // Check that the buffer isn't empty and
     // that num is less than number of added elements
-    if ((size == 0) || (size <= num)) {
-        goto fail;
+    if ((sizeint2 == 0u) || (size <= num)) {
```
Comment (sramrajkar) [inline app/app_src/circular_buffer.c:117]: Flip constant and variable so that the comparision is 0u == sizeint2

File: app/app_src/crc.c [hunk 18]
```diff
@@ -16,24 +16,25 @@ Copyright 2020 Samuel Ramrajkar
...
-    
-    //CRC_Start();
+
     CRCCON0bits.EN = 1;
     CRCCON0bits.CRCGO = 1;
+    
     for(uint16_t i = 0; i < len; i++ )
     {
-        while(CRCCON0bits.FULL); //Wait if the module is busy
+        while(CRCCON0bits.FULL == 1){} //Wait if the module is busy
```
Comment (sramrajkar) [inline app/app_src/crc.c:32]: flip constant and hardware bit

File: app/app_src/crc.c [hunk 19]
```diff
@@ -16,24 +16,25 @@ Copyright 2020 Samuel Ramrajkar
...
         CRC_8BitDataWrite(*ptr);  
         ptr++;
     }
     //Now wait for the result to be computed
-    while(CRCCON0bits.BUSY); //Wait if the module is busy
+    while(CRCCON0bits.BUSY == 1){} //Wait if the module is busy
```
Comment (sramrajkar) [inline app/app_src/crc.c:37]: same as above

File: app/app_src/led.c [hunk 20]
```diff
@@ -29,41 +29,43 @@ void ledInit(void)
     
     //Now load the version blink activity on the queue
     //Initial off period
-    queueLedEvent(0,2000);
+    uint8_t ledEventStatus = queueLedEvent(0,2000);
```
Comment (sramrajkar) [inline app/app_src/led.c:32]: Lets talk about this on the call

File: app/app_src/led.c [hunk 21]
```diff
@@ -76,18 +78,28 @@ void handle_led_events(void)
             struct ledEvent temp;
             CircularBufferPopFront(&ledtasksbuf,&temp);
             //Check if it was a zero time event
-            if(0 == temp.eventDuration){
+            int Eduration = (int)temp.eventDuration;
+            if(0 == Eduration){
```
Comment (sramrajkar) [inline app/app_src/led.c:82]: whats the error here?

File: app/app_src/led.c [hunk 22]
```diff
@@ -76,18 +78,28 @@ void handle_led_events(void)
...
+                else{
+                    LEDcond =  true;
+                }
+                if(LEDcond){
+                    LED_SetLow();
+                }
+                else{
+                    LED_SetHigh();           
+                }
+                set_timer0base(&ledtimer, temp.eventDuration);
```
Comment (sramrajkar) [inline app/app_src/led.c:101]: let talk about this

File: app/app_src/uart_default_control.c [hunk 23]
```diff
@@ -44,23 +44,27 @@ void uart_default_engine(void){
             }
             break;
         case SWITCH_TO_DEFAULT:
-            set_parity(UART_9BIT_EVEN_PARITY);
-            set_uart_baud(UART_BAUD_19200);
+            {
+                uint8_t parity_status = set_parity(UART_9BIT_EVEN_PARITY);
+                uint8_t uart_baud_status = set_uart_baud(UART_BAUD_19200);   
+            }
```
Comment (sramrajkar) [inline app/app_src/uart_default_control.c:50]: lets talk about this

File: app/app_src/uart_default_control.c [hunk 24]
```diff
@@ -44,23 +44,27 @@ void uart_default_engine(void){
...
 #ifdef MBRTU
             eMBInit( MB_RTU, MB_RTU_ADDR_MAX, 0, UART_BAUD_19200, 
                                              UART_9BIT_EVEN_PARITY);
 #endif
             uart_default_state_var = WAIT_GPIO_GO_HIGH;
             break;
+            
         case WAIT_GPIO_GO_HIGH:
-            if(BLEN_GetValue()){
+            if(BLEN_GetValue() && true){
```
Comment (sramrajkar) [inline app/app_src/uart_default_control.c:59]: Lets talk about this

File: app/app_src/uart_default_control.c (hunk 12 above)
Comment (sramrajkar) [reply app/app_src/uart_default_control.c:85]: Lets talk about this

File: app/app_src/led.c (hunk 20 above)
Comment (sramrajkar) [reply app/app_src/led.c:32]: (void) queueLedEvent(0,2000);

File: app/app_src/crc.c [hunk 25]
```diff
@@ -16,24 +16,25 @@ Copyright 2020 Samuel Ramrajkar
 #include "mcc.h"
 #include "crc.h"
 
-uint16_t crc16_app(void* dptr, uint16_t len, uint16_t seed){
-    uint8_t* ptr = (uint8_t*)dptr;
+uint16_t crc16_app(uint8_t* dptr, uint16_t len, uint16_t seed){
+    uint8_t* ptr = dptr;
     uint16_t result;
+    bool fullbool;
```
Comment (sramrajkar) [inline app/app_src/crc.c:22]: unused variable

File: app/app_src/uart_default_control.c (hunk 24 above)
Comment (sramrajkar) [reply app/app_src/uart_default_control.c:59]: 0 != BLEN_GetValue()

**Comment by atharva-18** [comment]
@sramrajkar I have addressed the remaining issues.

**Pull request closed without merging**